### PR TITLE
[Youtube] Fixed problem with new youtube player

### PR DESCRIPTION
**Pull request by peet1993**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Discussion for the fix can be found in #56
This fixes occasional "Unable to extract video data" caused by a change of YouTube's player.

### CODE REVIEW

File: youtube_dlc/extractor/youtube.py [hunk 1]
```diff
@@ -1784,21 +1796,25 @@ def extract_embedded_config(embed_webpage, video_id):
...
+                            # Rental video is not rented but preview is available (e.g.
+                            # https://www.youtube.com/watch?v=yYr8q0y5Jfg,
+                            # https://github.com/ytdl-org/youtube-dl/issues/10532)
+                            if not video_info and args.get('ypc_vid'):
+                                return self.url_result(
+                                    args['ypc_vid'], YoutubeIE.ie_key(), video_id=args['ypc_vid'])
+                            if args.get('livestream') == '1' or args.get('live_playback') == 1:
+                                is_live = True
+                            if not player_response:
+                                player_response = extract_player_response(args.get('player_response'), video_id)
```
Comment (abayochocoball) [inline youtube_dlc/extractor/youtube.py:1814]: Is any of this stuff going to be relevant with the new player? Same with the captions stuff above. I don't know what this stuff is for but will more changes have to be made later to support this stuff in the new player?

Comment (peet1993) [reply youtube_dlc/extractor/youtube.py:1814]: Good question. I am pretty confident that the automatic caption stuff works, all other code is from previous youtube player versions and basically does the same in a different way.

The stuff you marked seems to be about "Is this a live stream?", "Is this a previewable payed video?" and a third thing I do not understand (the "url_encoded_fmt_stream_map").

I checked a livestream, that part is fine. It is checked again from the video_details (which is extracted from the player_response) in line 1976. This even happens for the "old" player.
```
     if is_live is None:
            is_live = bool_or_none(video_details.get('isLive'))
```
For payed video with a preview, yt-dlc just says "This video requires payment to watch" (I used a movie as an example). I'm not quite sure when this code is used, and the example in the comment is down :/

**Comment by blackjack4494** [comment]
It appears to be working for those few videos I tested.

File: youtube_dlc/extractor/youtube.py (hunk 1 above)
Comment (blackjack4494) [reply youtube_dlc/extractor/youtube.py:1814]: As there haven't been any more comments the last 3 days I assume the conversation is resolved.

**Review by blackjack4494 — APPROVED** [review]

**Pull request merged**